lib/shared-method.js

```diff
@@ -269,35 +302,55 @@ function coerceAccepts(uarg, desc) {
     throw new badArgumentError(name + ' must be a number');
   }
 
-  // convert strings
-  if (actualType === 'string' && targetType !== 'any' && actualType !== targetType) {
-    // JSON.parse can throw, so catch this error.
-    try {
-      uarg = convertValueToTargetType(uarg, targetType);
-    } catch (e) {
-      var message = util.format('invalid value for argument \'%s\' of type ' +
-        '\'%s\': %s. Received type was %s. Error: %s',
-        name, targetType, uarg, typeof uarg, e.message);
-      throw new badArgumentError(message);
-    }
-  }
   return uarg;
 }
 
-function convertValueToTargetType(value, targetType) {
+function convertToBasicRemotingType(type) {
+  if (Array.isArray(type)) {
+    return type.map(convertToBasicRemotingType);
+  }
+
+  if (typeof type === 'object') {
+    type = type.modelName || type.name;
+  }
+
+  type = String(type).toLowerCase();
+  switch (type) {
+    case 'string':
+    case 'number':
+    case 'date':
+    case 'boolean':
+    case 'buffer':
+    case 'object':
+    case 'any':
+      return type;
+    default:
+      // custom types like MyModel
+      return 'object';
```

Comment:
> This might break complex type coercion. 

Can you please describe such scenario, so that I can add a unit-test for that?

> Perhaps you can check that list here?

To be honest, I feel it's out of scope of this pull request. If the data read from the request is an object, and the argument type is object-like (either object, any, or custom type from LoopBack), then I think it's ok to proceed.

If we start checking constructor names, then we might break applications that are working now.